src/CarbonAware/test/extensions/EmissionsDataExtensionsTests.cs

```diff
@@ -0,0 +1,112 @@
+
+using CarbonAware.Model;
+using CarbonAware.Extensions;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+
+
+namespace CarbonAware.Extensions.Tests;
+
+public class EmissionsDataExtensionsTests
+{
+    // Test class sets these fields in [SetUp] rather than traditional class constructor.
+    #pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    private IEnumerable<EmissionsData> data;
+    private TimeSpan dataDuration;
+    private DateTimeOffset dataStartTime;
+    private string dataLocation;
+    #pragma warning restore CS8618
+
+    [SetUp]
+    public void Setup()
+    {
+        this.dataDuration =  TimeSpan.FromMinutes(5);
+        this.dataStartTime = new DateTimeOffset(2020, 1, 1, 0, 0, 0, TimeSpan.Zero);
+        this.dataLocation = "A";
+        this.data = new List<EmissionsData>()
```

Comment:
nit: Seems like this is not needed? I see that all the time there are a new list created per test. This is going to be created always for all the tests.

Reply:
Good point!  I changed this to only be called once rather than per-test.